msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -1834,6 +1835,45 @@ public void onError(MsalException exception) {
         }
     }
 
+    public void acquireTokenWithDeviceCode(@NonNull List<String> scopes, @NonNull final DeviceCodeFlowCallback callback, @Nullable ClaimsRequest claimsRequest, @Nullable UUID correlationId) {
+        DeviceCodeFlowParameters.Builder builder = new DeviceCodeFlowParameters.Builder();
+        DeviceCodeFlowParameters deviceCodeFlowParameters =
+                builder.withScopes(scopes)
+                        .withClaims(claimsRequest)
+                        .withCorrelationId(correlationId)
+                        .build();
+
+        final DeviceCodeFlowCommandParameters commandParameters = CommandParametersAdapter
+                .createDeviceCodeFlowWithClaimsCommandParameters(
+                        mPublicClientConfiguration,
+                        mPublicClientConfiguration.getOAuth2TokenCache(),
+                        deviceCodeFlowParameters);
+
+        final DeviceCodeFlowCommandCallback deviceCodeFlowCommandCallback = getDeviceCodeFlowCommandCallback(callback);
+
+        try {
+            final DeviceCodeFlowCommand deviceCodeFlowCommand = new DeviceCodeFlowCommand(
+                    commandParameters,
+                    MSALControllerFactory.getDefaultController(
+                            mPublicClientConfiguration.getAppContext(),
+                            commandParameters.getAuthority(),
+                            mPublicClientConfiguration
+                    ),
+                    deviceCodeFlowCommandCallback,
+                    PublicApiId.DEVICE_CODE_FLOW_WITH_CALLBACK
```

Comment:
>DEVICE_CODE_FLOW_WITH_CALLBACK [](http://example.com/codeflow?start=32&length=30)

new API id here?
(Unless we are deprecating the below one)

Reply:
done, thanks!